lib/pbench/server/api/resources/datasets_compare.py

```diff
@@ -90,6 +110,16 @@ def _get(
                 )
             )
 
+            # Determine the dataset benchmark and check consistency
+            benchmark = DatasetsCompare.get_benchmark_name(dataset)
+            if not benchmark_choice:
+                benchmark_choice = benchmark
+            elif benchmark != benchmark_choice:
+                raise APIAbort(
```

Comment:
It's a bit of a nicety, but I would do these comparisons in the opposite order:
```python
            benchmark = DatasetsCompare.get_benchmark_name(dataset)
            if benchmark != benchmark_choice:
                if not benchmark_choice:
                    benchmark_choice = benchmark
                else:
                    raise APIAbort(
```
That is, if `benchmark` _is_ equal to `benchmark_choice` then we don't need to bother about anything.  If `benchmark_choice` is `None` that just makes it unlikely that `benchmark` will be equal to it, and we'll proceed to the next step, which is to decide whether to report an error or settle on the benchmark choice.  So, on the first dataset, we'll still do two comparisons, but on every subsequent dataset we'll do only one.  (Granted, there aren't likely to be very many datasets, so we're not likely to save very many comparisons, but....)

Reply:
This code is unchanged. It's even a bit inaccurate that GitHub is highlighting it as if it moved, because (in terms of my actual editing sequence) I didn't even move it: I moved the permission check that GitHub thinks stayed where it was. 😆